include/xtensor/xgenerator.hpp

```diff
@@ -100,6 +101,7 @@ namespace xt
         using iterable_base = xconst_iterable<self_type>;
         using inner_shape_type = typename iterable_base::inner_shape_type;
         using shape_type = inner_shape_type;
+        using dynamic_shape_type = xt::dynamic_shape<std::size_t>;
```

Comment:
`using dynamic_shape_type = xt::dynamic_shape<typename shape_type::value_type>;` so all the shape types have the same value type.